### ISSUE TITLE
[Question] Vous faites une seconde passe pour les role ARIA?

### ISSUE THREAD

**Issue by JayPanoz** [civodulab/epubtools#21]
La colonne “elements allowed on” du mapping `epub:type` → `role` a été ajoutée récemment, suite à https://github.com/w3c/epubcheck/issues/1010. 

Et autant que je m'en souvienne, ça a été remonté dès le module npm de Rodebert promu sur twitter, mais l’idée était que c'était la responsabilité d’un autre module de repasser derrière pour vérifier l'élément sur lequel le `role` a été ajouté et corriger si besoin.

J'ai ouvert l'issue comme “heads up” vu que ce sera reporté par EPUBCheck 4.2.0 mais si c'est déjà géré, pas de souci.

**Comment by JayPanoz** [comment]
Note: ça devrait expliciter l'issue #19 – je me suis tapé la même dans Blitz. 

Dans mon cas, le `role="directory"` devait être sur la liste imbriquée dans `nav`, pas directement sur `nav`. 

Mais a priori, avec l'update du mapping, vous pourriez en avoir davantage à l'avenir, d'où la nécessité de checker si l'élément est OK pour l’ajout. Ce serait peut-être une bonne idée de créer des tests automatisés pour ça, du coup.

**Comment by garconvacher** [comment]
Non, pas de vérification poussée. Basiquement, les roles DPUB sont ajoutés dans les balises contenant un epub:type s'il n'y en a pas, sauf si c'est dans un `<nav>` landmarks (voir issues https://github.com/w3c/epubcheck/issues/965 et https://github.com/daisy/ace/issues/220).
Mais ça pourrait être pas mal de vérifier ceux qui sont déjà présents où il faut pas. T'en penses quoi @civodulab ?

**Comment by garconvacher** [comment]
@JayPanoz Tu parles de [ton tweet](https://twitter.com/JiminyPan/status/1107268770589011968) pour le nmp de Roderbert ?

**Comment by JayPanoz** [comment]
Non c'était un autre beaucoup plus ancien – que je ne retrouve plus. Il date de l'année dernière mais il n'apporte pas vraiment dans ce contexte. 

En gros on avait eu une réponse ”c'est cool mais il faut faire attention avec les `role` parce qu’on peut pas les placer n’importe où, contrairement aux `epub:type` donc ce serait bien que le module le fasse.”

Et c’est parti en explication du modèle npm i.e. “logiquement c’est à un autre module derrière de prendre la main et replacer les `role` où il faut, etc.”

**Comment by civodulab** [comment]
Aucun soucis, je peux vérifier le bon emplacement des `role`

**Comment by garconvacher** [comment]
[Tweet](https://twitter.com/JiminyPan/status/970618482307018753) trouvé je pense !

**Comment by JayPanoz** [comment]
@garconvacher yep, c'est celui-là. GG
